internal/repositories/rewards.go

```diff
@@ -0,0 +1,218 @@
+		VehicleID:               reward.VehicleID,
+	}
+}
+
+func (r *Repository) GetEarningsByVehicleID(ctx context.Context, tokenID int) (*gmodel.VehicleEarnings, error) {
+	pHelp := helpers.PaginationHelper[RewardsCursor]{}
+
+	type rewardStats struct {
+		TokenSum   types.NullDecimal `boil:"token_sum"`
+		TotalCount int               `boil:"total_count"`
+	}
+	var stats rewardStats
+
+	limit := new(int)
+	*limit = 100
+
+	err := models.Rewards(
+		qm.Select(
+			fmt.Sprintf(
+				`sum(%s + %s + %s) as token_sum`,
+				models.RewardColumns.StreakEarnings, models.RewardColumns.AftermarketEarnings, models.RewardColumns.SyntheticEarnings,
+			),
+			"count(*) as total_count",
+		),
+		models.RewardWhere.VehicleID.EQ(tokenID),
+	).Bind(ctx, r.pdb.DBS().Reader, &stats)
+	if err != nil {
+		return nil, err
+	}
+
+	queryMods := []qm.QueryMod{
+		models.RewardWhere.VehicleID.EQ(tokenID),
+		qm.OrderBy(models.RewardColumns.IssuanceWeek + " desc"),
+	}
+
+	queryMods = append(queryMods, qm.Limit(*limit))
+
+	rewards, err := models.Rewards(queryMods...).All(ctx, r.pdb.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	edges := make([]*gmodel.EarningsEdge, len(rewards))
+	nodes := make([]*gmodel.Earning, len(rewards))
+
+	for i, rw := range rewards {
+		earning := RewardToAPI(*rw)
+		cursor, err := pHelp.EncodeCursor(RewardsCursor{
+			Week:      earning.Week,
+			VehicleID: tokenID,
+		})
+		if err != nil {
+			return nil, err
+		}
+		edges[i] = &gmodel.EarningsEdge{
+			Node:   &earning,
+			Cursor: cursor,
+		}
+
+		nodes[i] = &earning
+	}
+
+	if len(edges) == 0 {
+		return &gmodel.VehicleEarnings{}, nil
+	}
+
+	endCursor, err := pHelp.EncodeCursor(RewardsCursor{
+		Week:      edges[len(edges)-1].Node.Week,
+		VehicleID: tokenID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	startCursor, err := pHelp.EncodeCursor(RewardsCursor{
+		Week:      edges[0].Node.Week,
+		VehicleID: tokenID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	earningsConn := &gmodel.EarningsConnection{
+		TotalCount: stats.TotalCount,
+		PageInfo: &gmodel.PageInfo{
+			EndCursor:   &endCursor,
+			StartCursor: &startCursor,
+		},
+		Edges: edges,
+		Nodes: nodes,
+	}
+
+	return &gmodel.VehicleEarnings{
+		TotalTokens: dbtypes.NullDecimalToInt(stats.TokenSum),
+		History:     earningsConn,
+		VehicleID:   tokenID,
+	}, nil
+}
+
+func (r *Repository) PaginateVehicleEarningsByID(ctx context.Context, vehicleEarnings *gmodel.VehicleEarnings, first *int, after *string, last *int, before *string) (*gmodel.EarningsConnection, error) {
+	pHelp := helpers.PaginationHelper[RewardsCursor]{}
+	limit, err := pHelp.ValidateFirstLast(first, last, maxPageSize)
+	if err != nil {
+		return nil, err
+	}
+
+	if vehicleEarnings == nil || vehicleEarnings.History == nil {
+		return &gmodel.EarningsConnection{}, nil
+	}
+
+	erningsConnectionCopy := *vehicleEarnings.History
+	earningIdx := map[string]int{}
+	for idx, val := range erningsConnectionCopy.Edges {
```

Comment:
I don't think this interplay between cursors and limits is going to work out. We took the limit in the parent resolver before ever seeing the cursor. Like suppose our list is

* Week 4
* Week 3
* Week 2
* Week 1

We get arguments of `first: 2` and `after: week 3`. In the parent resolver I think we're going to slap a `LIMT 2` on the query and get

* Week 4
* Week 3

But the request calls for 

* Week 2
* Week 1

How will this work out?